webapp/src/test/java/com/box/l10n/mojito/service/cache/DatabaseCacheTest.java

```diff
@@ -0,0 +1,228 @@
+        Assertions.assertEquals(VALUE, databaseCache.lookup(KEY));
+        Assertions.assertEquals(VALUE2, databaseCache.lookup(KEY2));
+
+        databaseCache.evict(KEY);
+
+        Assertions.assertNull(databaseCache.lookup(KEY));
+        Assertions.assertEquals(VALUE2, databaseCache.lookup(KEY2));
+    }
+
+    @Transactional
+    @Test
+    public void testClearWorksCorrectly() {
+        databaseCache.put(KEY, VALUE);
+        databaseCache.put(KEY2, VALUE2);
+        Assertions.assertEquals(VALUE, databaseCache.lookup(KEY));
+        Assertions.assertEquals(VALUE2, databaseCache.lookup(KEY2));
+
+        databaseCache.clear();
+
+        Assertions.assertNull(databaseCache.lookup(KEY));
+        Assertions.assertNull(databaseCache.lookup(KEY2));
+        Assertions.assertEquals(-0, ((ApplicationCacheRepository) databaseCache.getNativeCache()).findAll().size());
+    }
+
+    @Transactional
+    @Test
+    public void testEvictEntryOnDeserializationFailure() {
+        SerializationPair<Object> defaultSerializationPair = new DefaultSerializationPair();
+        SerializationPair<Object> mockSerializationPair = mock(DefaultSerializationPair.class);
+
+        when(mockSerializationPair.write(ArgumentMatchers.any())).thenReturn(defaultSerializationPair.write(VALUE));
+
+        when(mockSerializationPair.read(any())).thenThrow(new RuntimeException());
+        databaseCacheConfiguration.setValueSerializationPair(mockSerializationPair);
+
+        DatabaseCache databaseCacheWithMockedDeserializer = new DatabaseCache("databaseCacheWithMockedDeserializer", databaseCacheConfiguration);
+        databaseCacheWithMockedDeserializer.put(KEY, VALUE);
+        Assertions.assertNull(databaseCacheWithMockedDeserializer.lookup(KEY));
+    }
+
+    @Transactional
+    @Test
+    public void testExpiryDateLookup() {
+        Duration ttl = Duration.ofSeconds(2);
+        databaseCacheConfiguration.setTtl(ttl);
+        DatabaseCache databaseCacheWith1SecondTTL = new DatabaseCache("databaseCacheWith1SecondTTL", databaseCacheConfiguration);
+
+        databaseCacheWith1SecondTTL.put(KEY, VALUE);
+        Assertions.assertEquals(VALUE, databaseCacheWith1SecondTTL.lookup(KEY));
+
+        try {
+            Thread.sleep(ttl.toMillis());
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        Assertions.assertNull(databaseCacheWith1SecondTTL.lookup(KEY));
+    }
+
+    @Test
+    public void testExpiryBasedEviction() throws InterruptedException {
+        Duration ttl = Duration.ofSeconds(1);
+        databaseCacheConfiguration.setTtl(ttl);
+        DatabaseCache databaseCacheWithAggressiveEviction = new DatabaseCache("databaseCacheWithAggressiveEviction", databaseCacheConfiguration);
+
+        // This test is not supported on H2
+        if (dbUtils.isHsql()) {
+            return;
+        }
+
+        putKeyValue(databaseCacheWithAggressiveEviction, KEY, VALUE);
+        Thread.sleep(ttl.toMillis());
```

Comment:
Sleeping in test eventually makes the test suite slow. Can we avoid?

Reply:
Avoided now